scripts/linux_installer.py

```diff
@@ -0,0 +1,92 @@
+        self.THYRA_APP_FILENAME = "thyra-app"
+        self.MKCERT_FILENAME = "mkcert"
+
+        self.THYRA_SERVER_URL = "https://github.com/massalabs/thyra/releases/latest/download/thyra-server_linux_amd64"
+        self.THYRA_APP_URL = "https://github.com/massalabs/Thyra-Menu-Bar-App/releases/latest/download/ThyraApp_linux-amd64"
+        self.MKCERT_URL = "https://dl.filippo.io/mkcert/latest?for=linux/amd64"
+
+    def configureNetworkManager(self):
+        logging.info("Configuring NetworkManager...")
+        self.executeCommand("sudo cp /etc/NetworkManager/NetworkManager.conf /etc/NetworkManager/NetworkManager.conf.backup_thyra_install", True)
+        stdout, _stderr = self.executeCommand("grep '^dns=' /etc/NetworkManager/NetworkManager.conf", True, allow_failure=True)
+        dns = stdout.removeprefix("dns=").strip()
+        if dns == "dnsmasq":
+            logging.info("NetworkManager is already configured")
+        elif dns == "":
+            # add dns=dnsmasq after [main] line
+            self.executeCommand("sudo sed -i 's/^\[main\]$/\[main\]\\ndns=dnsmasq/' /etc/NetworkManager/NetworkManager.conf", True)
+        else:
+            self.executeCommand("sudo sed -i 's/^dns=.*/dns=dnsmasq/g' /etc/NetworkManager/NetworkManager.conf", True)
+
+    def configureDNSMasq(self):
+        logging.info("Configuring DNSMasq...")
+
+        self.executeCommand("sudo mkdir -p /etc/NetworkManager/dnsmasq.d", True)
+        self.executeCommand("sudo bash -c 'echo 'address=/.massa/127.0.0.1' > /etc/NetworkManager/dnsmasq.d/massa.conf'", True)
+        self.executeCommand("sudo mv /etc/resolv.conf /etc/resolv.conf.backup_thyra_install", True)
+        self.executeCommand("sudo ln -s /run/NetworkManager/resolv.conf /etc/resolv.conf", True)
+        self.executeCommand("sudo systemctl restart NetworkManager", True)
+
+    def setupDNS(self):
+        stdout, _stderr = self.executeCommand("sudo lsof -i :53", True, allow_failure=True)
+        runningDNS = ""
+        if stdout:
+            runningDNS = stdout.splitlines()[1].split()
+            runningDNS = runningDNS[0]
+
+        if runningDNS == "":
+            logging.info("Installing dnsmasq...")
+            self.executeCommand("sudo apt-get install -y dnsmasq", True)
+            self.configureNetworkManager()
+            self.configureDNSMasq()
+        elif runningDNS == "dnsmasq":
+            logging.info("dnsmasq is already installed")
+            self.configureDNSMasq()
+        elif runningDNS == "systemd-r":
+            logging.warning("Your computer is using systemd-resolved for DNS. Thyra needs dnsmasq to redirect .massa domains to localhost.")
+            user_answer = input("Do you want to install dnsmasq and configure it to redirect .massa domains to localhost? [y/n] ")
+            user_answer = user_answer.lower()
+            if user_answer == "y" or user_answer == "yes":
+                self.configureNetworkManager()
```

Comment:
in that case maybe we should make sure that dndmasq is installed?

```suggestion
                self.executeCommand("sudo apt-get install -y dnsmasq", True)
                self.configureNetworkManager()
```